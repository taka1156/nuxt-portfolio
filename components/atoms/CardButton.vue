<template functional>
  <button class="btn" :class="data.class || ''" v-on="listeners">
    <slot />
  </button>
</template>

<style scoped>
.btn {
  display: block;
  font-size: 1em;
  width: 75%;
  margin: 10px auto;
  padding: 10px;
  background-color: white;
  color: gray;
  border: 1px solid gray;
  border-radius: 5px;
}

.btn:active,
.btn:hover {
  background-color: gray;
  color: white;
}
</style>
