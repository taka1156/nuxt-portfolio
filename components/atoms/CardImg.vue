<template functional>
  <img v-lazy="props.cardImg" class="card__img" />
</template>

<style scoped>
.card__img {
  margin: 0 auto;
  height: 100%;
  width: 100%;
}

/* lazy-load */
img[lazy='loaded'] {
  animation: 1s anime-opacity both;
}
</style>
