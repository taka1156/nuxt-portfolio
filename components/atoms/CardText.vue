<template functional>
  <p class="card__text">
    {{ props.cardText }}
  </p>
</template>

<style scoped>
.card__text {
  width: 98%;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
  border: solid 0.5px lightgray;
}
</style>
