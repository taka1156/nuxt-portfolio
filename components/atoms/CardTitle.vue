<template functional>
  <h2 class="card__title">{{ props.cardTitle }}</h2>
</template>

<style scoped>
.card__title {
  font-size: 1.75em;
  line-height: 0em;
  text-align: center;
}
</style>
