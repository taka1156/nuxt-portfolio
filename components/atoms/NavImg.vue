<template functional>
  <img :src="props.img" class="nav__icon" />
</template>

<style scoped>
.nav__icon {
  height: 50px;
  width: 50px;
}
</style>
