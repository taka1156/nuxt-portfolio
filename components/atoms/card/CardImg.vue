<template functional>
  <img
    v-lazy="props.cardImg"
    :alt="props.imgAlt"
    class="card-img"
    :class="[props.isPortfolio ? 'card-img--lg' : 'card-img--sm']"
  />
</template>

<style scoped>
.card-img {
  display: block;
  margin: 0 auto;
  padding: 5px;
  box-sizing: border-box;
}

.card-img--sm {
  height: 50%;
  width: 50%;
}

.card-img--lg {
  height: 100%;
  width: 100%;
}
</style>
