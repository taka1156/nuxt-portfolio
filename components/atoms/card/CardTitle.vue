<template>
  <div>
    <h2 class="card-title">
      <!-- @slot カードタイトル -->
      <slot />
    </h2>
  </div>
</template>

<style scoped>
.card-title {
  font-size: 1.3em;
  line-height: 0em;
  text-align: center;
}
</style>
