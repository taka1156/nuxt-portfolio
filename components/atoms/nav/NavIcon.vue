<template functional>
  <div class="nav-icon">
    <span
      class="nav-icon__top"
      :class="[props.isOpen ? 'nav-icon__top--open' : 'nav-icon__top--close']"
    ></span>
    <span
      class="nav-icon__middle"
      :class="{ 'nav-icon__middle--fade': props.isOpen }"
    ></span>
    <span
      class="nav-icon__bottom"
      :class="[props.isOpen ? 'nav-icon__bottom--open' : 'nav-icon__bottom--close']"
    ></span>
  </div>
</template>

<style scoped>
.nav-icon {
  position: relative;
  top: 0;
  right: 0;
  height: 50px;
  width: 50px;
  cursor: pointer;
}

span {
  display: block;
  height: 2px;
  width: 70%;
  margin: auto;
  position: absolute;
  top: calc((100% - 4px) / 2);
  left: 0;
  right: 0;
  background-color: #fff;
  border-radius: 1px;
}

.nav-icon__top--close {
  transform: translateY(-10px);
}

.nav-icon__bottom--close {
  transform: translateY(10px);
}

.nav-icon__top--open {
  animation: top 0.6s ease;
  transform: rotate(-45deg);
}

.nav-icon__middle--fade {
  animation: middle 0.6s ease;
  opacity: 0;
}

.nav-icon__bottom--open {
  animation: bottom 0.6s ease;
  transform: rotate(45deg);
}

@keyframes top {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(45deg);
  }
}

@keyframes middle {
  from {
    opacity: 1;
  }

  to {
    opacity: 0;
  }
}

@keyframes bottom {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(-45deg);
  }
}
</style>
