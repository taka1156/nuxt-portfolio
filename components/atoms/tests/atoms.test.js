import { shallowMount } from '@vue/test-utils';
import NavIcon from '../NavIcon.vue';
import NavImg from '../NavImg.vue';
import NavLogo from '../NavLogo.vue';
import NavText from '../NavText.vue';
import CardTitle from '../CardTitle.vue';
import CardImg from '../CardImg.vue';
import CardText from '../CardText.vue';
import CardButton from '../CardButton.vue';

// 備考: 全て関数型コンポーネント

// NavIcon
describe('NavIcon', () => {
  const navIcon = propsData => {
    return shallowMount(NavIcon, {
      propsData: {
        ...propsData,
      },
    });
  };

  it('開いた時(閉じるアイコン表示)', () => {
    const wrapper = navIcon({ isOpen: true });
    const TOP = wrapper.find('.nav-icon__top');
    const MIDDLE = wrapper.find('.nav-icon__middle');
    const BOTTOM = wrapper.find('.nav-icon__bottom');
    expect(TOP.classes()).toContain('nav-icon__top--open');
    expect(MIDDLE.classes()).toContain('nav-icon__middle--fade');
    expect(BOTTOM.classes()).toContain('nav-icon__bottom--open');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('閉じる時(開くアイコン表示)', () => {
    const wrapper = navIcon({ isOpen: false });
    const TOP = wrapper.find('.nav-icon__top');
    const MIDDLE = wrapper.find('.nav-icon__middle');
    const BOTTOM = wrapper.find('.nav-icon__bottom');
    expect(TOP.classes()).toContain('nav-icon__top--close');
    expect(MIDDLE.classes()).not.toContain('nav-icon__middle--fade');
    expect(BOTTOM.classes()).toContain('nav-icon__bottom--close');
    expect(wrapper.html()).toMatchSnapshot();
  });
});

// NavImg
describe('NavImg', () => {
  it('値がDOMに反映されているか', () => {
    const dummyImg = {
      img: 'http://placehold.jp/150x150.png',
      alt: 'ダミー画像',
    };
    const navImg = shallowMount(NavImg, {
      propsData: {
        navImg: dummyImg.img,
        imgAlt: dummyImg.alt,
      },
    });
    const imgTag = navImg.find('img');
    expect(imgTag.attributes().src).toBe(dummyImg.img);
    expect(imgTag.attributes().alt).toBe(dummyImg.alt);
    expect(navImg.html()).toMatchSnapshot();
  });
});

// NavLogo
describe('NavLogo', () => {
  it('値がDOMに反映されているか', () => {
    const dummyLogo = 'ダミーロゴ';
    const navLogo = shallowMount(NavLogo, {
      slots: {
        default: dummyLogo,
      },
    });
    expect(navLogo.text()).toBe(dummyLogo);
    expect(navLogo.html()).toMatchSnapshot();
  });
});

// NavText
describe('NavText', () => {
  it('値がDOMに反映されているか', () => {
    const dummyText = 'ダミーテキスト';
    const navText = shallowMount(NavText, {
      slots: {
        default: dummyText,
      },
    });
    expect(navText.text()).toBe(dummyText);
    expect(navText.html()).toMatchSnapshot();
  });
});

describe('CardButton', () => {
  it('slotがDOMに反映されているか', () => {
    const dummyButtonText = 'ダミーボタンテキスト';
    const cardButton = shallowMount(CardButton, {
      slots: {
        default: dummyButtonText,
      },
    });
    const btnTag = cardButton.find('button');
    expect(btnTag.text()).toBe(dummyButtonText);
    expect(btnTag.html()).toMatchSnapshot();
  });
});

describe('CardImg', () => {
  it('値がDOMに反映されているか', () => {
    const dummyImg = {
      img: 'http://placehold.jp/150x150.png',
      alt: 'ダミー画像',
    };
    const navImg = shallowMount(CardImg, {
      propsData: {
        cardImg: dummyImg.img,
        imgAlt: dummyImg.alt,
      },
    });
    const imgTag = navImg.find('img');
    // v-lazyからsrcを取り出せない
    // expect(imgTag.attributes().src).toBe(dummyImg.img);
    expect(imgTag.attributes().alt).toBe(dummyImg.alt);
    expect(navImg.html()).toMatchSnapshot();
  });
});

describe('CardText', () => {
  it('値がDOMに反映されているか', () => {
    const dummyText = 'ダミーテキスト';
    const cardText = shallowMount(CardText, {
      slots: {
        default: dummyText,
      },
    });
    const pTag = cardText.find('p');
    expect(pTag.text()).toBe(dummyText);
    expect(pTag.html()).toMatchSnapshot();
  });
});

describe('CardTitle', () => {
  it('値がDOMに反映されているか', () => {
    const dummyTitle = 'ダミータイトル';
    const cardTitle = shallowMount(CardTitle, {
      slots: {
        default: dummyTitle,
      },
    });
    const hTag = cardTitle.find('h2');
    expect(hTag.text()).toBe(dummyTitle);
    expect(cardTitle.html()).toMatchSnapshot();
  });
});
