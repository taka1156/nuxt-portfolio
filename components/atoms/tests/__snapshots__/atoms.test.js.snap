// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardButton slotがDOMに反映されているか 1`] = `"<button class=\\"btn\\">ダミーボタンテキスト</button>"`;

exports[`CardImg 値がDOMに反映されているか 1`] = `"<img alt=\\"ダミー画像\\" class=\\"card__img\\">"`;

exports[`CardText 値がDOMに反映されているか 1`] = `"<p class=\\"card__text\\">ダミーテキスト</p>"`;

exports[`CardTitle 値がDOMに反映されているか 1`] = `"<h2 class=\\"card__title\\">ダミータイトル</h2>"`;

exports[`NavIcon 閉じる時(開くアイコン表示) 1`] = `"<div class=\\"border__group\\"><span class=\\"border__top border__top--close\\"></span> <span class=\\"border__middle\\"></span> <span class=\\"border__bottom border__bottom--close\\"></span></div>"`;

exports[`NavIcon 開いた時(閉じるアイコン表示) 1`] = `"<div class=\\"border__group\\"><span class=\\"border__top border__top--open\\"></span> <span class=\\"border__middle border__middle--fade\\"></span> <span class=\\"border__bottom border__bottom--open\\"></span></div>"`;

exports[`NavImg 値がDOMに反映されているか 1`] = `"<img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー画像\\" class=\\"nav__icon\\">"`;

exports[`NavLogo 値がDOMに反映されているか 1`] = `"<a href=\\"/\\" class=\\"nav__logo\\">ダミーロゴ</a>"`;

exports[`NavText 値がDOMに反映されているか 1`] = `"<div class=\\"nav__text\\">ダミーテキスト</div>"`;
